### PR TITLE
classwise and mean IOU

### PR DESCRIPTION
**Pull request by victorlazio109**
- Added classwise and mean IOU functionalities.

### CODE REVIEW

File: utils/metrics.py [hunk 1]
```diff
@@ -71,10 +73,23 @@ def report_classification(pred, label, batch_size, metrics_dict, ignore_index=-1
...
 
 
-def iou(pred, target, batch_size, metrics_dict):
-    """Calculate the intersection over union (or Jaccard index) between two datasets.
-    The Jaccard distance (or dissimilarity) would be 1-iou.
-    """
-    iou = jaccard_similarity_score(target.cpu(), pred.cpu(), normalize=True)
-    metrics_dict['iou'].update(iou, batch_size)
-    return metrics_dict
+def iou(pred, label, batch_size, num_classes, metric_dict, only_present=True):
```
Comment (mpelchat04) [inline utils/metrics.py:76]: Awesome! well done!  
Have you noticed some overhead caused by the time it takes to compute the IoU?

Comment (victorlazio109) [reply utils/metrics.py:76]: Very negligible at the test stage, if it is computed all to frequently at the validation set then we may notice some overhead. At the moment it is easy to live with.

File: utils/metrics.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
-from sklearn.metrics import classification_report, jaccard_similarity_score
```
[the remark this answers is not in the document]
Comment (victorlazio109) [reply utils/metrics.py:1]: @ymoisan jaccard score/index == IOU.  It is useful and can be used however to calculate the classwise IOU correctly it had to be done manually. You cannot pass the num_classes into the function.

File: conf/config_ci_segmentation_local.yaml [hunk 3]
```diff
@@ -52,12 +74,15 @@ training:
     rotate_limit: 45
     rotate_prob: 0.5
     hflip_prob: 0.5
+  normalization:
```
Comment (CharlesAuthier) [inline conf/config_ci_segmentation_local.yaml:77]: Can you add some comments on the meaning of using it please

Comment (victorlazio109) [reply conf/config_ci_segmentation_local.yaml:77]: sure

File: inference.py [hunk 4]
```diff
@@ -231,7 +231,7 @@ def main(params):
 
     default_working_folder = Path(params['inference']['state_dict_path']).parent.joinpath(f'inference_{num_bands}bands')
     working_folder = Path(get_key_def('working_folder', params['inference'], default_working_folder)) # TODO: remove working_folder parameter in all templates
-    Path.mkdir(working_folder, exist_ok=True)
+    working_folder.mkdir(parents=True, exist_ok=True)
```
[the remark this answers is not in the document]
Comment (victorlazio109) [reply inference.py]: resolved, reverted change.
